datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -132,12 +104,60 @@ public SdxRecoveryResponse triggerRecovery(SdxCluster sdxCluster, SdxRecoveryReq
         }
     }
 
-    private SdxCluster getOldCluster(SdxCluster newCluster) {
-        return sdxClusterRepository.findByAccountIdAndEnvCrnAndDeletedIsNullAndDetachedIsTrue(
-                newCluster.getAccountId(), newCluster.getEnvCrn()
-        ).orElseThrow(notFound(
+    private SdxCluster getOldClusterErrorIfNotFound(SdxCluster newCluster) {
+        return getOldCluster(newCluster).orElseThrow(notFound(
                 "detached SDX cluster",
                 "Env CRN: " + newCluster.getEnvCrn() + ", Account ID: " + newCluster.getAccountId()
         ));
     }
+
+    private Optional<SdxCluster> getOldCluster(SdxCluster newCluster) {
+        Optional<SdxCluster> oldCluster = sdxClusterRepository.findByAccountIdAndEnvCrnAndDeletedIsNullAndDetachedIsTrue(
+                newCluster.getAccountId(), newCluster.getEnvCrn()
+        );
+        return (oldCluster.isPresent() && !oldCluster.get().getClusterName().equals(newCluster.getClusterName())) ?
+                oldCluster : Optional.empty();
+    }
+
+    private String getReasonNonRecoverable(DatalakeStatusEnum status) {
+        String reason = "";
+        if (status.equals(RUNNING)) {
```

Comment:
in case of every enum comparison, you should compare like this: `RUNNING.equals(status)`. In this case service won't fail even if status is null somehow

Reply:
Makes sense, thank you!